### ISSUE TITLE
`npm install [module]` deletes `node_modules` in symlinked modules

### ISSUE THREAD

**Issue by doowb** [npm/npm#17183]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [x] npm is producing an incorrect install.
  - [x] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### How can the CLI team reproduce the problem?

From project "a":
 - `npm install` install current dependencies for project "a"
 - `npm link` link project "a" to be used in project "b"

From project "b":
 - `npm link a` link "a" into project "b".
 - `npm install foo` install another modules from npm into project "b"

The last step removes the `node_modules` folder from project "a".

### supporting information:

 - `npm -v` prints:
   5.0.3
 - `node -v` prints:
   v6.10.1
 - `npm config get registry` prints:
   https://registry.npmjs.org
 - Windows, OS X/macOS, or Linux?:
   macOS Sierra v10.12.1

**Comment by stq** [comment]
We have a linked subproject dependency in node_modules, and each time when someone installs new package, it gets unlinked - have to link it again. Very unhandy, please fix or add option to control it.

**Comment by scottmas** [comment]
Yes. This. This is utterly clobbering my custom mono-repo setup. Also, it's worth noting that you don't even need to install a specific module. Running `npm install` in a project that has symlinked node_modules will clobber the node_modules of the linked module.

**Comment by doublerebel** [comment]
Dupe of #16788 (thanks to list of known issues for npm 5 at #16991)

**Comment by onury** [comment]
this is very annoying. I have 5 packages linked in a main project. if I install or remove a single module it will mess up the node_modules of all dependency linked modules.

**Comment by scottmas** [comment]
Yes. It essentially makes NPM link unusable
